Add clara-shikhelman:-unjamming-lightning to bitcoinplusplus/layer-2 (#371)
bitcointranscripts/bitcointranscripts: bitcoinplusplus/layer-2/clara-shikhelman:-unjamming-lightning.md

---

> title: "Clara Shikhelman: Unjamming Lightning"

Reviewer: There is no need to add the name of the speaker in the title, we already have it as metadata

transcript_by: NeroCherubino via review.btctranscripts.com
media: https://www.youtube.com/watch?v=S7ZH4hr2FlA
tags: ["lightning","attacks"]
speakers: ["Clara Shikhelman"]
categories: ["conference"]
date: 2023-07-01
---

**Introduction**

I'm Clara, I'm from Chaincode Labs, and I want to talk to you about an ongoing project that tries to solve the jamming problem in the Lightning Network.
We'll start from what it is and then we're trying to get it.

**Lightning Network Routing**

A very quick reminder to all of you about Lightning Network routing.

> Alice wants to route by Bob to Charlie and we have HTCLC for atomic swaps. The way it goes pretty much Alice talks let's say a hundred satoshis between herself and Bob telling him "if you'll get some secret it will be all yours" and then the instructions to Bob that the next hop is Charlie.

Reviewer: Here, the word "locks" was wrongly transcribed as "talks".
Also, notice that I removed "let's say" from the sentence and the I added an additional word (is) in parenthesis to make the sentence more understandable.
```suggestion
Alice wants to route by Bob to Charlie and we have HTCLs for atomic swaps. The way it goes pretty much Alice locks a hundred satoshis between herself and Bob telling him "if you'll get some secret it will be all yours" and then the instructions to Bob (is) that the next hop is Charlie.
```

Bob locks again funds between himself and Charlie saying "okay give me the secret you'll get the funds".
Charlie gives the secret, takes the funds, Bob with the secret goes to Alice, claiming the funds and Bob also can charge a fee when this succeeds.

But what happens when Charlie doesn't give the secret?

We have this funds locked and they're just hanging there for a while, two weeks, then Alice gets her money back and Bob gets his money back, but then for quite a while we have this unusable funds in the channels, which is not fantastic.

**Intro the Jamming**

> Even worse than that is that if we have a malicious actor, for example, we have the original channels between Alice, Bob, and Charlie, and then somebody, our beloved jammer, is trying to stop this chance from working, they can open channels to Alice and Charlie and then start routing a payment from themselves to Alice, to Bob, to Charlie and back to themselves.

Reviewer: Here, "channels" was wrongly transcribed as "chance"
```suggestion
Even worse than that is that if we have a malicious actor, for example, we have the original channels between Alice, Bob, and Charlie, and then somebody, our beloved jammer, is trying to stop this channels from working, they can open channels to Alice and Charlie and then start routing a payment from themselves to Alice, to Bob, to Charlie and back to themselves.
```

Locking all funds and everybody is waiting for a jammer to release the funds, but he has no intention of doing it and as you can see the direction from Bob to Charlie is completely stuck.

> No fans can move this way unless something happens from the other direction.

Reviewer: Here, "funds" was wrongly transcribed as "fans"
```suggestion
No funds can move this way unless something happens from the other direction.
```
